### PR TITLE
Code review fixes - fix creating config.ini file

### PR DESCRIPTION
**Pull request by mikelik**
I have tested that `config.ini` file is being created properly in docker.

### CODE REVIEW

File: src/dune/docker.py [hunk 1]
```diff
@@ -98,7 +99,10 @@ def rm_file(self, file_name):
         self.execute_cmd(['rm', '-rf', file_name])
 
     def write_file(self, file_name, body):
-        self.execute_cmd(['echo', body, '>', file_name])
+        with tempfile.NamedTemporaryFile(mode='w+',) as tmp:
+            tmp.write(body)
+            tmp.flush()
+            self.cp_from_host(tmp.name, file_name)
```
Comment (ScottBailey) [inline src/dune/docker.py:105]: Does Windows allow the copy while tmp remains open? If not, I think we need to add the param `delete=False` and manage close and deletion ourselves.

**Review by ScottBailey — CHANGES_REQUESTED** [review]
2 comments to address: 
- can Windows copy an open tempfile?
- remove or comment extra print command

File: src/dune/dune.py [hunk 2]
```diff
@@ -130,9 +130,11 @@ def start_node(self, nod, snapshot=None):
...
             current_ver = self.get_current_nodeos_version()
             config_str = get_config_ini(current_ver[0], current_ver[1], current_ver[2])
-            self._docker.write_file("config.ini", config_str)
+            self._docker.write_file("/app/config.ini", config_str)
             nod.set_config('/app/config.ini')
-        
+
+        print(nod.config_dir())
+
         print(nod.config_dir())
```
[the remark this answers is not in the document]
Comment (mikelik) [reply src/dune/dune.py:138]: yes, merging issue - thanks

File: src/dune/docker.py (hunk 1 above)
Comment (mikelik) [reply src/dune/docker.py]: Excellent catch, thanks!
Indeed this is the issue on Windows. Fixed in the latest commit.

Additionally I noticed the tests are not passing on Windows, so I created separate issues for that: https://github.com/AntelopeIO/DUNE/issues/186

File: src/dune/dune.py [hunk 3]
```diff
@@ -105,7 +105,7 @@ def create_node(self, nod):
 
     def get_current_nodeos_version(self):
         stdout, stderr, exit_code = self._docker.execute_cmd(['nodeos', '--version'])
-        return stdout[1:].split('.-')
+        return re.split(r"[.-]", stdout[1:-1])
```
Comment (ScottBailey) [inline src/dune/dune.py:108]:
```suggestion
        return re.split(r"[.-+]", stdout[1:-1])
```

After commenting about semver prerelease (`-`), I later realized  build (`+`) would be appropriate to split on as well. Think of this as a nice to have, not a must have.

File: src/dune/dune.py [hunk 4]
```diff
@@ -130,14 +130,12 @@ def start_node(self, nod, snapshot=None):
...
-                config_file.write(config_ini)
-                config_file.close()
-                self._docker.cp_from_host(config_file.name, "/app/config.ini")
-                os.remove(config_file.name)
+            config_str = get_config_ini(self._docker.get_arch(), current_ver[0], current_ver[1], current_ver[2])
+            self._docker.write_file("/app/config.ini", config_str)
             nod.set_config('/app/config.ini')
 
+
+
```
Comment (ScottBailey) [inline src/dune/dune.py:138]: 3 blank lines seems like 1 too mant? Nice to fix, not necessary to fix.

Comment (mikelik) [reply src/dune/dune.py]: Right, done

File: src/dune/dune.py (hunk 3 above)
Comment (mikelik) [reply src/dune/dune.py]: `.-+` gives error `re.error: bad character range .-+ at position 1`
but `+.-` seem to work which is interesting.
Anyway I have added it

File: tests/common.py [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
 
 # Find path for tests:
 TEST_PATH = os.path.dirname(os.path.abspath(__file__))
 
 # Set path for executable:
-DUNE_EXE = os.path.split(TEST_PATH)[0] + "/dune"
+if platform.system() == 'Windows':
+    DUNE_EXE = os.path.split(TEST_PATH)[0] + r"\dune.bat"
+else:
+    DUNE_EXE = os.path.split(TEST_PATH)[0] + "/dune"
```
Comment (ScottBailey) [inline tests/common.py:11]: Would this be better? I think it would.
```python
DUNE_EXE = os.path.join( os.path.split(TEST_PATH)[0] , "dune")
```

**Review by ScottBailey — CHANGES_REQUESTED** [review]
Please address setting DUNE_EXE in `tests/common.py`.

Feel free to address or ignore other comments.

**Review by ScottBailey — APPROVED** [review]

**Pull request merged**